web_bench/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+use js_sys::Date;
+use wasm_bindgen::prelude::*;
+#[cfg(feature = "parallel")]
+pub use wasm_bindgen_rayon::init_thread_pool;
+
+#[wasm_bindgen]
+pub fn generate_proof() -> JsValue {
+    let relation = JfRelation::Withdraw;
+    let srs = relation.generate_srs();
+    let (pk, _) = relation.generate_keys(&srs);
+
+    let start = Date::now();
+    relation.generate_proof(pk);
+    (Date::now() - start).into()
+}
+
+use ark_ff::PrimeField;
+use jf_primitives::{
+    merkle_tree::{
+        prelude::RescueSparseMerkleTree, MerkleCommitment, MerkleTreeScheme,
+        UniversalMerkleTreeScheme,
+    },
+    pcs::prelude::UnivariateUniversalParams,
+};
+use jf_relations::{
+    shielder_types::{compute_note, convert_array},
+    withdraw::{WithdrawPrivateInput, WithdrawPublicInput, WithdrawRelation},
+    Curve, Proof as JfProof, ProvingKey as JfProvingKey, Relation as _,
+    VerifyingKey as JfVerifyingKey,
+};
+use num_bigint::BigUint;
+
+pub enum JfRelation {
+    Withdraw,
+}
+
+impl JfRelation {
+    pub fn generate_srs(&self) -> UnivariateUniversalParams<Curve> {
+        let rng = &mut jf_utils::test_rng();
+        jf_relations::generate_srs(17_000, rng).unwrap()
+    }
+
+    pub fn generate_keys(
+        &self,
+        srs: &UnivariateUniversalParams<Curve>,
+    ) -> (JfProvingKey<Curve>, JfVerifyingKey<Curve>) {
+        WithdrawRelation::generate_keys(srs).unwrap()
+    }
+
+    pub fn generate_proof(&self, pk: JfProvingKey<Curve>) -> JfProof<Curve> {
+        let rng = &mut jf_utils::test_rng();
+        let relation = jf_relation();
+        relation.generate_proof(&pk, rng).unwrap()
+    }
+}
+
+fn jf_relation() -> WithdrawRelation {
```

Comment:
Slight code duplication. You could

- make this public in the `relations` crate 
 - hide it behind a feature flag 
 - use it here